File: SLR.py
#Data Preprocessing

#importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing the dataset
dataset = pd.read_csv('Salary_data.csv')

#Splitting into dependent and independent matrices
X = dataset.iloc[: , :-1].values
Y= dataset.iloc[:, 1].values

#Splitting the data into train and test models
from sklearn.model_selection import train_test_split
X_train,X_test,Y_train,Y_test = train_test_split(X,Y,test_size = 1/3,random_state = 0)

#Feature scaling is not included since the Simple Linear Regression algorithm does it on its own if necessary

#Training the model
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,Y_train)

#Predicting the test results
Y_pred = regressor.predict(X_test)

#Visualizing the training set results
plt.scatter(X_train,Y_train,color = 'red')
plt.plot(X_train,regressor.predict(X_train),color = 'blue')
plt.label('Experience vs Salary(Training Set)')
plt.xlabel('Experience')
plt.ylabel('Salary')
plt.show()

#Visualizing the test set results
plt.scatter(X_test,Y_test,color = 'red')
plt.plot(X_train,regressor.predict(X_train),color = 'blue')
plt.label('Experience vs Salary(Test Set)')
plt.xlabel('Experience')
plt.ylabel('Salary')
plt.show()

File: SLR.R
#Simple Linear Regression

#Removing the previous data
rm(list = ls())

#Importing the data
dataset = read.csv('Salary_Data.csv')

#Splitting the data into train and test models
library(caTools)
set.seed(123)
?sample.split()
split = sample.split(dataset$Salary,SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset,split == FALSE)

#Fitting Simple Linear Regression model to the training set
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

#Used to check the model  
summary(regressor)

#Predicting the Test Set Results
y_pred = predict(regressor, newdata = test_set)

#Visualising the training_set
install.packages('ggplot2')
library('ggplot2')

ggplot() + 
  geom_point(aes(x = training_set$YearsExperience,y = training_set$Salary),colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor,newdata = training_set)),colour = 'blue') +
  ggtitle('Salary vs YearsExperience in R(training set)') +
  xlab('Years of Experience') + 
  ylab('Salary')

#Visualising the test_set

ggplot() +
  geom_point(aes(x= test_set$YearsExperience,y = test_set$Salary),colour = 'red') +
  geom_line(aes(x=training_set$YearsExperience,y=predict(regressor,newdata = training_set)),colour = 'blue') +
  ggtitle('Salary vs Experience in R(test set)') + 
  xlab('Years of Experience') +
  ylab('Salary')
